docs/user/spark/sparksql.rst

```diff
@@ -279,3 +281,114 @@ write to a distributed file system like HDFS.
 
     For this to work, the Data Frame should have a geometry field, meaning its schema should have a ``StructField`` that
     is one of the JTS geometry types provided by GeoMesa.
+
+Using GeoMesa SparkSQL with Apache Sedona
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+GeoMesa SparkSQL can work seamlessly with `Apache Sedona <http://sedona.apache.org/>`_. You can enable this feature by
+specifying ``geomesa.use.sedona`` system property to ``true`` when submitting your Spark job. Please make sure that
+Apache Sedona JAR ``sedona-python-adapter-${spark-version}_${scala-version}-${sedona-version}.jar`` is submitted with
+``--jars`` option:
+
+.. code-block:: shell
+
+   spark-submit --conf "spark.driver.extraJavaOptions=-Dgeomesa.use.sedona=true" \
+                --conf "spark.executor.extraJavaOptions=-Dgeomesa.use.sedona=true" \
+                --jars /path/to/geomesa-spark-runtime-jar.jar,/path/to/sedona-python-adapter-jar.jar \
+                ...
+
+There are several configs to take care of when creating Spark session object:
+
+.. code-block:: scala
+
+    val spark = SparkSession.builder().
+      // ...
+      config("spark.serializer", "org.apache.spark.serializer.KryoSerializer").
+      config("spark.kryo.registrator", classOf[GeoMesaSparkKryoRegistrator].getName).
+      config("spark.geomesa.sedona.udf.prefix", "sedona_").
```

Comment:
would it make sense to have this be the default value if nothing is specified for the sedona prefix?

Reply:
I want to riff on this idea for a minute.  To clarify, I want to discuss what I am about to suggest as an idea; we should think about it and reach a decision first.

Given @Kontinuation's point about wanting to test with and without Sedona, I wonder if it'd be worth it to just generate GeoMesa-only Spark runtime jars AND GeoMesa+Sedona Spark runtime jars.

The Sedona project cannot bundle GeoTools due to Apache Software Foundation rules.  The Eclipse Foundation permits GeoMesa to bundle GeoTools (by a board-level exemption!).  Given that context, we could basically provide dependencies included options where things just work out of the box.

Thoughts?